DllImportGenerator/Ancillary.Interop/MarshalEx.cs

```diff
@@ -0,0 +1,29 @@
+
+using System.Reflection;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Marshalling helper methods that will likely live in S.R.IS.Marshal
+    /// when we integrate our APIs with dotnet/runtime.
+    /// </summary>
+    public static class MarshalEx
```

Comment:
Can't we just make these extension methods on `Marshal`? That way we can simply add the support in the runtime and it "just works"?

Reply:
You can't add extension static methods (and as a result you can't add extension methods to static classes). C# doesn't support that.